### PR TITLE
refactor(new-market-agreement): add agreement to v7 preview step

### PR DESCRIPTION
**Pull request by jaredvu**
Pull NewMarketAgreement into a separate `view` to be used in Dialog and directly in `v7/NewMarketPreviewStep`.

---

## Views

* New: `<NewMarketAgreement>`
  * Add Agreement form to it's own view

* `<NewMarketPreviewStep>` , `<NewMarketAgreementDialog>`
  * Use `<NewMarketAgreement>` view
 
 | Old flow | New flow |
 |---|---|
 | <video src="https://github.com/user-attachments/assets/319a413a-fa47-42c4-8346-74549c74d9df" /> | <video src="https://github.com/user-attachments/assets/66efc7fe-86c1-4d04-96b3-e3917f10f22b" /> |

* `<LaunchMarketSidePanel>`
  * Update copy to use `USDC` icon

## Functions

* `lib/tradingView/dydxfeed`
  * Use `symbol` to index the `lastBarsCache`

* `lib/tradingView/utils`
  * update `full_name` to use displayable ticker in the TradingView chart
  * Screen Shot 2024-09-12 at 11 17 39 AM

---

### CODE REVIEW

File: src/lib/tradingView/dydxfeed/index.ts [hunk 1]
```diff
@@ -248,7 +248,7 @@ export const getDydxDatafeed = (
       onRealtimeCallback,
       subscribeUID,
       onResetCacheNeededCallback,
-      lastBar: lastBarsCache.get(`${symbolInfo.full_name}/${RESOLUTION_MAP[resolution]}`),
+      lastBar: lastBarsCache.get(`${symbolInfo.symbol}/${RESOLUTION_MAP[resolution]}`),
```
Comment (tinaszheng) [inline src/lib/tradingView/dydxfeed/index.ts:251]: i dont think `.symbol` exists? also lol at having `any` types up there lol

File: src/views/forms/NewMarketForm/NewMarketAgreement.tsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        id="acknowledgement-checkbox"
+        label={stringGetter({ key: STRING_KEYS.I_HAVE_READ_AND_AGREE })}
+      />
+      <div tw="grid grid-cols-[1fr_2fr] gap-1">
+        <Button action={ButtonAction.Base} onClick={() => onCancel()}>
+          {stringGetter({ key: STRING_KEYS.CANCEL })}
+        </Button>
+        <Button
+          action={ButtonAction.Primary}
+          onClick={() => {
```
Comment (tinaszheng) [inline src/views/forms/NewMarketForm/NewMarketAgreement.tsx:60]:
```suggestion
          onClick={onAccept}
```

File: src/views/forms/NewMarketForm/NewMarketAgreement.tsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+      </p>
+
+      <Checkbox
+        checked={hasAcknowledged}
+        onCheckedChange={setHasAcknowledged}
+        id="acknowledgement-checkbox"
+        label={stringGetter({ key: STRING_KEYS.I_HAVE_READ_AND_AGREE })}
+      />
+      <div tw="grid grid-cols-[1fr_2fr] gap-1">
+        <Button action={ButtonAction.Base} onClick={() => onCancel()}>
```
Comment (tinaszheng) [inline src/views/forms/NewMarketForm/NewMarketAgreement.tsx:55]:
```suggestion
        <Button action={ButtonAction.Base} onClick={onCancel}>
```

File: src/lib/tradingView/dydxfeed/index.ts (hunk 1 above)
Comment (jaredvu) [reply src/lib/tradingView/dydxfeed/index.ts:251]: Ah will change to `ticker` For some reason I thought `symbolInfo` was the return from `getSymbol` function. Thanks for catching

**Review by tinaszheng — APPROVED** [review]

File: src/lib/tradingView/dydxfeed/index.ts (hunk 1 above)
Comment (tinaszheng) [reply src/lib/tradingView/dydxfeed/index.ts:251]: i guess `full_name` doesnt exist either 😆

**Review by tinaszheng — APPROVED** [review]

**Pull request merged**